### PR TITLE
pypi release workflow

### PR DESCRIPTION
**Pull request by yinweisu**
Issue #, if available:

Description of changes:
* Modified based on the workflow being used by main AG repo

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pypi_release.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python
+      uses: actions/setup-python@v1
```
Comment (tonyhoo) [inline .github/workflows/pypi_release.yml:20]: I thought it is not the latest stable version anymore. Can you check if we need to upgrade here (e.g. for security reasons)?

Comment (yinweisu) [reply .github/workflows/pypi_release.yml:20]: You correct. These are not the latest. Let me update them to be latest and we probably want to do the same to AG workflow as well

Comment (tonyhoo) [reply .github/workflows/pypi_release.yml]: Thanks

**Review by tonyhoo — APPROVED** [review]

File: .github/workflows/pypi_release.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  release:
+    types: [created]
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3.3.0
```
Comment (gradientsky) [inline .github/workflows/pypi_release.yml:18]: Can we just do `@v3` as it is highlighted in tutorials on the [action github](https://github.com/actions/checkout)?

File: .github/workflows/pypi_release.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Set up Python
+      uses: actions/setup-python@v4.5.0
```
Comment (gradientsky) [inline .github/workflows/pypi_release.yml:20]: ditto: `@v4`

File: .github/workflows/pypi_release.yml (hunk 2 above)
Comment (yinweisu) [reply .github/workflows/pypi_release.yml:18]: Well, if you click use latest button on this page: https://github.com/marketplace/actions/checkout you get what I put there. I think it doesn't matter to much anyways...

File: .github/workflows/pypi_release.yml (hunk 3 above)
Comment (yinweisu) [reply .github/workflows/pypi_release.yml:20]: ditto

**Review by gradientsky — APPROVED** [review]

**Pull request merged**